File: flutter/vex_owl_admin/lib/core/services/template_service.dart
import 'package:flutter_riverpod/flutter_riverpod.dart';
import '../../core/constants/api_constants.dart';
import '../../core/utils/api_client.dart';
import '../../data/models/template_models.dart';
import '../../data/models/query_models.dart';

final templateServiceProvider = Provider<TemplateService>((ref) {
  return TemplateService(ref.read(apiClientProvider));
});

class TemplateService {
  final ApiClient _apiClient;

  TemplateService(this._apiClient);

  Future<List<TemplateEntity>> queryTemplates(QueriesPageRequest request) async {
    final response = await _apiClient.post<List<TemplateEntity>>(
      ApiConstants.templateQuery,
      data: request.toJson(),
      parser: (data) => (data as List).map((e) => TemplateEntity.fromJson(e)).toList(),
    );
    return response;
  }

  Future<TemplateEntity> getTemplate(String id) async {
    final response = await _apiClient.get<TemplateEntity>(
      '${ApiConstants.template}/$id',
      parser: (data) => TemplateEntity.fromJson(data),
    );
    return response;
  }

  Future<TemplateEntity> getTemplateByCode(String code) async {
    final response = await _apiClient.get<TemplateEntity>(
      '${ApiConstants.templateByCode}/$code',
      parser: (data) => TemplateEntity.fromJson(data),
    );
    return response;
  }

  Future<TemplateEntity> createTemplate(Map<String, dynamic> data) async {
    final response = await _apiClient.post<TemplateEntity>(
      ApiConstants.template,
      data: data,
      parser: (data) => TemplateEntity.fromJson(data),
    );
    return response;
  }

  Future<TemplateEntity> updateTemplate(String id, TemplateUpdateRequest request) async {
    final response = await _apiClient.put<TemplateEntity>(
      '${ApiConstants.template}/$id',
      data: request.toJson(),
      parser: (data) => TemplateEntity.fromJson(data),
    );
    return response;
  }

  Future<void> deleteTemplate(String id) async {
    await _apiClient.delete('${ApiConstants.template}/$id');
  }
}

File: ow-admin/src/core/services/templateService.ts
import { apiClient } from '../utils/apiClient';
import { ApiConstants } from '../constants/apiConstants';
import { TemplateEntity, TemplateUpdateRequest } from '../../data/models/templateModels';
import { QueriesPageRequest } from '../../data/models/queryModels';

export class TemplateService {
  async queryTemplates(request: QueriesPageRequest): Promise<TemplateEntity[]> {
    return apiClient.post<TemplateEntity[]>(ApiConstants.templateQuery, request);
  }

  async getTemplate(id: string): Promise<TemplateEntity> {
    return apiClient.get<TemplateEntity>(`${ApiConstants.template}/${id}`);
  }

  async getTemplateByCode(code: string): Promise<TemplateEntity> {
    return apiClient.get<TemplateEntity>(`${ApiConstants.templateByCode}/${code}`);
  }

  async createTemplate(data: Record<string, unknown>): Promise<TemplateEntity> {
    return apiClient.post<TemplateEntity>(ApiConstants.template, data);
  }

  async updateTemplate(id: string, request: TemplateUpdateRequest): Promise<TemplateEntity> {
    return apiClient.put<TemplateEntity>(`${ApiConstants.template}/${id}`, request);
  }

  async deleteTemplate(id: string): Promise<void> {
    return apiClient.delete<void>(`${ApiConstants.template}/${id}`);
  }
}

export const templateService = new TemplateService();